code/_onclick/observer.dm

```diff
@@ -14,7 +14,7 @@
 
 	// Otherwise jump
 	else if(A.loc)
-		loc = get_turf(A)
+		Move(get_turf(A.loc))

```

Comment:
As the local expert on the matter yes ghosts do not call Move(). They should, but I also presume it'll break their movement to do this. Just make it a forceMove() for now and be done with it.